### PR TITLE
feat(menu-item): Update spacing and icon layout

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #7105 

## Summary
Updates Menu Item to reflect Figma designs for complex combinations of props:
Screenshot 2023-07-26 at 3 22 06 PM

### CODE REVIEW

File: packages/calcite-components/src/components/menu-item/menu-item.tsx
Comment (driskull) [inline packages/calcite-components/src/components/menu-item/menu-item.tsx:450]: Does this need a `key` attr?

**Review by driskull — APPROVED** [review]
👍

File: packages/calcite-components/src/components/menu-item/menu-item.tsx
Comment (macandcheese) [reply packages/calcite-components/src/components/menu-item/menu-item.tsx:450]: Moved this to its own render method and added a key.

File: packages/calcite-components/src/components/menu-item/menu-item.tsx
Comment (jcfranco) [inline packages/calcite-components/src/components/menu-item/menu-item.tsx:446]: Does the icon scale need to adjust based on the menu item's scale?

File: packages/calcite-components/src/components/menu-item/menu-item.tsx
Comment (jcfranco) [inline packages/calcite-components/src/components/menu-item/menu-item.tsx:445]: Can `key` be `"hrefIcon"` or something along those lines? I find it hard to understand the association between the dropdown action CSS class and the href icon.

File: packages/calcite-components/src/components/menu/menu.stories.ts
Comment (jcfranco) [inline packages/calcite-components/src/components/menu/menu.stories.ts:86]: Nitpick: empty tags should avoid whitespace.

File: packages/calcite-components/src/components/menu/menu.stories.ts
Comment (jcfranco) [inline packages/calcite-components/src/components/menu/menu.stories.ts:56]: Nitpick: extra space before the HTML.

**Review by jcfranco — APPROVED** [review]
Noice! ✨👨‍🍳🥣✨

File: packages/calcite-components/src/components/menu-item/menu-item.tsx
Comment (macandcheese) [reply packages/calcite-components/src/components/menu-item/menu-item.tsx:445]: Wrong class used, fixed.

File: packages/calcite-components/src/components/menu-item/menu-item.tsx
Comment (macandcheese) [reply packages/calcite-components/src/components/menu-item/menu-item.tsx:446]: Menu / Menu Item don't currently have scale, but when added down the road this would change.

**Comment by macandcheese** [comment]
Ready to merge pending Chromatic approval for added and updated stories.

**Pull request merged**
